canvas_modules/common-canvas/webpack.config.js

```diff
@@ -37,68 +58,60 @@ module.exports = {
 		"common-canvas": "./src/index.js" // needs to be last to create correct combined css output
 	},
 	output: {
-		libraryTarget: "commonjs2",
 		filename: "[name].js",
-		path: path.join(__dirname, "/dist")
+		path: path.join(__dirname, "/dist"),
+		library: "[name]",
+		libraryTarget: "umd"
 	},
 	module: {
 		rules: [
-			{
-				enforce: "pre",
-				test: /\.js$/,
-				exclude: /node_modules\/intl-/,
-				loader: "source-map-loader"
-			},
 			{
 				test: /\.js(x?)$/,
 				exclude: /node_modules/,
 				loader: "babel-loader",
 				query: {
-					presets: ["react", "env"],
-					plugins: isCoverage ? ["istanbul"] : []
+					babelrc: false,
+					presets: ["@babel/preset-react", ["@babel/preset-env", { modules: false }]],
+					plugins: babelPlugins
 				}
 			},
 			{
 				test: /\.s*css$/,
-				use: ExtractTextPlugin.extract(
+				use: [
 					{
-						fallback: "style-loader",
-						use: [
-							{ loader: "css-loader" },
-							{ loader: "postcss-loader", options: { ident: "postcss", plugins: [require("autoprefixer")] } },
-							{ loader: "sass-loader", options: { includePaths: ["node_modules"] } }
-						]
+						loader: MiniCssExtractPlugin.loader,
+					},
+					{ loader: "css-loader" },
+					{ loader: "postcss-loader", options: { ident: "postcss", plugins: [require("autoprefixer")] } },
+					{ loader: "sass-loader",
+						options: {
+							sassOptions: {
+								includePaths: [".", "node_modules"]
+							}
+						}
 					}
-				)
+				]
 			},
 			{
 				test: /\.(woff|woff2|ttf|svg|png|eot)$/,
 				loader: "url-loader"
 			}
 		]
 	},
-	plugins: [
-		new webpack.DefinePlugin({
-			"process.env.NODE_ENV": "'production'"
-		}),
-		new webpack.optimize.UglifyJsPlugin({ sourceMap: true }), // minify everything
-		new webpack.optimize.AggressiveMergingPlugin(), // Merge chunk
-		new ExtractTextPlugin("common-canvas.css"),
-		new ExtractTextPlugin("common-canvas.min.css"),
-		new OptimizeCssAssetsPlugin({
-			assetNameRegExp: /\.min.css$/g,
-			cssProcessorOptions: { discardComments: { removeAll: true }, postcssZindex: { disable: true } }
-		}),
-		new BundleAnalyzerPlugin(
-			{ generateStatsFile: true, openAnalyzer: false }),
-		new webpack.ContextReplacementPlugin(/moment[/\\]locale$/, /en/)
-	],
 	resolve: {
 		modules: ["node_modules"],
 		extensions: [".js", ".jsx", ".json"]
 	},
-	externals: [nodeExternals(
-		{
-			whitelist: [/^d3.*$/]
-		})]
+	plugins: plugins,
+	performance: {
```

Comment:
This was added to help catch errors with importing modules incorrectly.  This will error the build if the module size is larger than 2 mb.